sql/insert.go

```diff
@@ -77,8 +77,17 @@ func (p *planner) Insert(n *parser.Insert) (planNode, error) {
 
 	for rows.Next() {
 		values := rows.Values()
-		if len(values) != len(cols) {
-			return nil, fmt.Errorf("invalid values for columns: %d != %d", len(values), len(cols))
+		for i := 0; i < len(values); i++ {
+			if col := cols[i]; values[i] == parser.DNull && !col.Nullable {
+				return nil, fmt.Errorf("null value in column %q violates not-null constraint", col.Name)
```

Comment:
I don't feel strongly (there was a question mark there). We've capitalized our SQL keywords elsewhere, though I'm not sure if we've done so in error messages.